[dynamo] log all graph breaks to graph_breaks logging artifact

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137244

We were previously not logging all graph breaks (e.g. data dependent jumps) to the graph_breaks logging artifact.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

williamwen42 committed:
[dynamo] log all graph breaks to graph_breaks logging artifact

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] log all graph breaks to graph_breaks logging artifact"


We were previously not logging all graph breaks (e.g. data dependent jumps) to the graph_breaks logging artifact.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

ezyang commented on test/dynamo/test_misc.py:
```diff
@@ -5795,7 +5795,9 @@ def g2(a, b):
         def count_graph_break_msgs(msgs):
-            return sum(msg.find("Graph break") != -1 for msg in msgs)
+            return sum(
+                "Graph break" in msg and "details suppressed" not in msg for msg in msgs
```
?

williamwen42 replied:
Filtering out details suppressed graph break messages (to comply with removing duplicate detailed graph break messages)

williamwen42 replied:
Simplified this check.

mlazos commented:
At a high level, why do we have separate graph break handling in those two places? 

What I'm thinking is that could we put the log graph break logic in `compile_subgraph` and check whether the compile reason is graph break? That would've prevented us from missing these other `compile_subgraph` calls.

williamwen42 commented:
@mlazos good point, let me try doing that.

williamwen42 committed:
Update on "[dynamo] log all graph breaks to graph_breaks logging artifact"


We were previously not logging all graph breaks (e.g. data dependent jumps) to the graph_breaks logging artifact.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

williamwen42 commented:
@mlazos I'm going to merge this PR for now and refactor in a followup PR.

williamwen42 commented:
@pytorchbot merge
